resharper/resharper-unity/src/Rider/UnitTesting/UnityTaskRunnerHostControllerExtension.cs

```diff
@@ -0,0 +1,111 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading.Tasks;
+using JetBrains.Application.Threading;
+using JetBrains.Diagnostics;
+using JetBrains.Lifetimes;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Host.Features.BackgroundTasks;
+using JetBrains.ReSharper.Host.Features.Unity;
+using JetBrains.ReSharper.UnitTestFramework;
+using JetBrains.ReSharper.UnitTestFramework.Launch;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.Rider.UnitTesting
+{
+    [SolutionComponent]
+    public class UnityTaskRunnerHostControllerExtension : ITaskRunnerHostControllerExtension
+    {
+        private static readonly Key<LifetimeDefinition> ourLifetimeDefinitionKey = new Key<LifetimeDefinition>("UnityTaskRunnerHostController.CancelPrepareForRun");
+        
+        private const string PluginName = "Unity plugin";
+        private const string NotAvailableUnityEditorMessage = "Unable to {0} tests: Unity Editor is not running";
+        private const string StartUnityEditorQuestionMessage = "To {0} unit tests, you should first run Unity Editor. Do you want to Start Unity {1} now?";
+
+        private readonly Lifetime myLifetime;
+        private readonly IUnityController myUnityController;
+        private readonly RiderBackgroundTaskHost myRiderBackgroundTaskHost;
+        private readonly IShellLocks myShellLocks;
+        private readonly IDictionary<string, string> myAvailableProviders;
+        private readonly object myStartUnitySync = new object();
+        private Task myStartUnityTask;
+
+        public UnityTaskRunnerHostControllerExtension(Lifetime lifetime,
+                                                      IShellLocks shellLocks,
+                                                      IUnityController unityController,
+                                                      RiderBackgroundTaskHost riderBackgroundTaskHost)
+        {
+            myLifetime = lifetime;
+            myShellLocks = shellLocks.NotNull();
+            myUnityController = unityController.NotNull();
+            myRiderBackgroundTaskHost = riderBackgroundTaskHost;
+            myStartUnityTask = Task.CompletedTask;
+            myAvailableProviders = new Dictionary<string, string>
+            { 
+                { WellKnownHostProvidersIds.RunProviderId, "Run" },
+                { WellKnownHostProvidersIds.DebugProviderId, "Debug" }
+            };
+        }
+
+        public bool IsApplicable(IUnitTestRun run)
+        {
+            var isUnity = myUnityController.IsUnityEditorUnitTestRunStrategy(run.RunStrategy);
+            return isUnity && myAvailableProviders.ContainsKey(run.HostController.HostId);
+        }
+
+        public ClientControllerInfo GetClientControllerInfo(IUnitTestRun run, ITaskRunnerHostController next) => null;
```

Comment:
if you don't need this method, can I remove it from the interface? I only added it because I saw that dotCover's variant of writing an extension was overriding `controllerInfo`. I would actually like to delete `GetClientControllerInfo` from 'ITaskRunnerHostContoller' altogether. This method is only related to a specific implementation of runners, and should not be in base interface.

Reply:
Yes, I agree, but I think that you mean delete GetClientControllerInfo from ITaskRunnerHostContoller**Extension**?

Reply:
Both :)

Reply:
Ok, I've understood